.github/workflows/node-zxc-build-release-artifact.yaml

```diff
@@ -115,119 +115,66 @@ permissions:
-
-          echo "branch-name=${BRANCH_NAME}" >>"${GITHUB_OUTPUT}"
-          echo "branch-name-lower=${BRANCH_NAME_LOWER}" >>"${GITHUB_OUTPUT}"
-          echo "branch-name-safe=${BRANCH_NAME_SAFE}" >>"${GITHUB_OUTPUT}"
-          echo "commit-prefix=${COMMIT_PREFIX}" >>"${GITHUB_OUTPUT}"
-          echo "commit-id=${{ github.sha }}" >>"${GITHUB_OUTPUT}"
-          echo "commit-id-short=$(echo "${{ github.sha }}" | cut -c1-8)" >>"${GITHUB_OUTPUT}"
-
-      - name: Verify Version Update (As Specified)
-        if: ${{ inputs.version-policy == 'specified' && !cancelled() && !failure() }}
-        run: |
-          VALID_VERSION="$(semver validate "${{ inputs.new-version }}")"
-
-          if [[ "${VALID_VERSION}" != "valid" ]]; then
-            echo "::error title=Version Error::The supplied new-version parameter (${{ inputs.new-version }}) is invalid and does not conform to the semantic versioning specifications."
-            exit 2
-          fi
-
       - name: Checkout Code
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-
-      - name: Setup Java
-        uses: actions/setup-java@8df1039502a15bceb9433410b1a100fbe190c53b # v4.5.0
         with:
-          distribution: ${{ inputs.java-distribution }}
-          java-version: ${{ inputs.java-version }}
+          fetch-depth: 0
 
-      - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@dbbdc275be76ac10734476cc723d82dfe7ec6eda # v3.4.2
+      - name: Setup Node.js
+        uses: actions/setup-node@0a44ba7841725637a19e28fa30b79a866c81b0a6 # v4.0.4
         with:
-          gradle-version: ${{ inputs.gradle-version }}
+          node-version: 'lts/*'
 
-      - name: Gradle Update Version (As Specified)
-        uses: gradle/actions/setup-gradle@dbbdc275be76ac10734476cc723d82dfe7ec6eda # v3.4.2
-        if: ${{ inputs.version-policy == 'specified' && !cancelled() && !failure() }}
-        with:
-          gradle-version: ${{ inputs.gradle-version }}
-          arguments: versionAsSpecified -PnewVersion=${{ inputs.new-version }} --scan
+      - name: Install dependencies
+        run: |
+          npm ci
+          npm install --save-dev \
+            @semantic-release/changelog \
+            @semantic-release/git \
+            conventional-changelog-conventionalcommits
 
-      - name: Gradle Update Version (Branch Commit)
-        uses: gradle/actions/setup-gradle@dbbdc275be76ac10734476cc723d82dfe7ec6eda # v3.4.2
-        if: ${{ inputs.version-policy != 'specified' && !cancelled() && !failure() }}
+      - name: Determine Version
+        id: semantic
+        uses: semantic-release/semantic-release@43df51b79ec9c8716b737804486b375459ea177e # v24.1.2
         with:
-          gradle-version: ${{ inputs.gradle-version }}
-          arguments: versionAsPrefixedCommit -PcommitPrefix=${{ steps.parameters.outputs.commit-prefix }} --scan
+          dry_run: true
```

Comment:
Will this ever successfully run if the `dry_run` field is set to `true`?

Reply:
I've revamped `node-zxc-build-release-artifact.yaml` and these steps are kept as they were.